radix-engine/src/transaction/transaction_executor.rs

```diff
@@ -667,16 +671,34 @@ where
             // Record final payments
             *fee_payments.entry(vault_id).or_default() += amount;
         }
+        // Free credit is locked first and thus used last
+        if free_credit.is_positive() {
```

Comment:
I wonder if we should include `used_free_credit` on the fee_summary?

I guess your feeling is maybe not, as it will always be zero for actual transactions?

Reply:
I don't have a strong opinion, as `total_cost() - total_payments()` would give the free credit amount.

Reply:
I will add a function and see how useful it is. :)

Reply:
I meant more as an actual field, but you're right, because it's calculated, it's probably not needed explicitly!